packages/notifier/src/publish-kit.js

```diff
@@ -178,14 +179,19 @@ harden(makePublishKit);
 /**
  * @param {object} [options]
  * @param {DurablePublishKitValueDurability & 'mandatory'} [options.valueDurability='mandatory']
+ * @param {import('@agoric/internal/src/callback.js').Callback<*>} [options.onAdvance] A direct callback
```

Comment:
"advance" makes sense internally to this function but is a new concept to the consumer. I would have expected `onPublish`, but I see that would omit `finish` and `fail`. One way through this is to have `publish` be the garden path "publish" and to consider `finish` and `fail` as also publishing values. Of course that would have its own confusion. 

Trying to think of a more obvious name for "any new value". Maybe `onChange`? `onNew`?

Reply:
Yeah, I had similar thoughts. How would you feel about `onUpdate`?

Reply:
perfect

Reply:
`onChange` and `onUpdate` are both fine with me, but I'd be ok with `onPublish` too. From what I understand `fail` won't cause a published value, and completion could be considered a final publish.

Reply:
I slightly prefer onPublish because it's a term the user already knows and they *should* understand that finish published and fail doesn't.

The confusion I was trying to avoid with alternatives to onPublish is really about the method names. I'd sooner rename finish() to publishFinal()

Reply:
`fail` _does_ advance the publish kit; it's an alternative termination to `finish`.

Reply:
When consuming the values through a subscriber, `fail` would result in a rejection, so by that logic, it's not observable as a published value, where both the values provided to `publish` and `finish` are.

Reply:
Right, but the new callback isn't a subscriber and will get called with an object encapsulating the terminal rejection.

Reply:
Oh I had missed that. Looks like it's updated with a rejected promise. We should not call the callback with either a value or a promise. Either we skip the rejection, or we wrap in a object like `PromiseSettlement`

Reply:
I'd prefer to keep the symmetry between `subscribeAfter` promise settlement and values sent to the callback. If we don't like that, my next higher preference would be not adding this functionality at all and instead handling it entirely by wrapping the `makeDurablePublishKit` calls such that values are sent to chain storage at the same place as they are sent to the publish kit.
```js
const makeDurablePublishKit = prepareDurablePublishKit(baggage, label);
const makeDurableChainStoragePublishKit = prepareDurableChainStoragePublishKit(
  makeDurablePublishKit,
  shared.publicMarshaller,
);
…
// methods of this `publisher` wrap those of a publisher returned from `makeDurablePublishKit`
// to write to `storageNode` *and* invoke the inner object method.
const { publisher, subscriber } = makeDurableChainStoragePublishKit(storageNode);
```

Reply:
Currently the `pipeToStorage` mechanism ignores failures. I understand we want to potentially support other use cases that care about the failure case.

An alternative is to always wrap the value given to the callback in a promise. a `Promise.resolve(resolution)` would work for that. My concern is having an argument that is sometimes a value, and sometimes a (rejected) promise.